### DISCUSSION TITLE
Guided mode

### DISCUSSION THREAD

**Discussion by ndit-dev** [danielmiessler/Fabric#771] [Ideas]
I use Fabric daily as my primary tool for interacting with various AI models. To streamline terminal usage, I've created a small Go program that acts as a terminal interface for Fabric. One crucial feature I've been missing is the ability to choose a pattern on the fly, eliminating the need to research a good pattern in advance.

To address this, I implemented a fuzzy finder that allows you to select a pattern without manually typing the pattern name.

Check it out [here](https://github.com/ndit-dev/FabricPrompt/blob/main/readme.md) or watch the demo to see how it works.

This project started as a quick hack for my personal use, but I believe it would be beneficial to incorporate a similar feature into the main Fabric project. Perhaps as a command-line switch that enables a "guided mode" for pattern selection.

**Comment by snafu4** [comment]
Been there, done that, got shot down:  https://github.com/danielmiessler/fabric/issues/318 and built a much better version for personal use.
